tasks/managed/enforce-labels/enforce-labels.yaml

```
@@ -158,3 +161,60 @@ spec:
                 fi
             fi
         done
+    - name: enforce-cpe-label
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      computeResources:
+        limits:
+          memory: 100Mi
+        requests:
+          memory: 100Mi
+          cpu: 10m
+      script: |
+        #!/usr/bin/env bash
+        set -ex
+        
+        DATA_FILE="$(params.dataDir)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+        
+        SNAPSHOT_FILE="$(params.dataDir)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+        
+        CPE_DATA=$(jq -r '.releaseNotes.cpe // empty' "${DATA_FILE}")
+        
+        if [[ -z "$CPE_DATA" ]]; then
+            echo "Error: The required static value 'releaseNotes.cpe' is missing or empty in the data file." >&2
+            exit 1
+        fi
+        
+        num_components=$(jq '.components | length' "${SNAPSHOT_FILE}")
+        
+        # Now iterate through components to check the label against the single fetched value
+        for ((i = 0; i < num_components; i++))
+        do
+            component=$(jq -c --argjson i "$i" '.components[$i]' "${SNAPSHOT_FILE}")
+            name=$(jq -r '.name' <<< "$component")
+        
+            # Fetch the value of the 'cpe' label from the component's metadata
+            cpe_label=$(jq -r '.metadata.labels? | .[] | select(.name == "cpe") | .value // empty' <<< "$component")
```

From AI:

> In jq, the ? suppresses errors on the key lookup (.metadata.labels?), but does not suppress errors from applying array operations (.[]) on a null value.

So maybe this? I mean, it should really be there, so it depends. But I think the goal of the `?` was to make it work even if it's not there. So if we want that, then maybe use the default empty list.
```suggestion
            cpe_label=$(jq -r '.metadata.labels // [] | .[] | select(.name == "cpe") | .value // empty' <<< "$component")
```

---

Modified.

---

Now the line is too long 😿

---

Fixed.